src/main/java/org/threeten/extra/chrono/InternationalFixedChronology.java

```diff
@@ -0,0 +1,431 @@
+     *
+     * @return the list of eras for the chronology, may be immutable, not null
+     */
+    @Override
+    public List<Era> eras () {
+        return Arrays.<Era> asList (InternationalFixedEra.values ());
+    }
+
+    //-----------------------------------------------------------------------
+
+    /**
+     * Gets the ID of the chronology - 'Ifc'.
+     * <p>
+     * The ID uniquely identifies the {@code Chronology}.
+     * It can be used to lookup the {@code Chronology} using {@link #of(String)}.
+     *
+     * @return the chronology ID - 'Ifc'
+     * @see #getCalendarType()
+     */
+    @Override
+    public String getId () {
+        return "Ifc";
+    }
+
+    /**
+     * Gets the calendar type of the underlying calendar system - 'ifc'.
+     * <p>
+     * The <em>Unicode Locale Data Markup Language (LDML)</em> specification
+     * does not define an identifier for the Pax calendar, but were it to
+     * do so, 'pax' is highly likely to be chosen.
+     *
+     * @return the calendar system type - 'ifc'
+     * @see #getId()
+     */
+    @Override
+    public String getCalendarType () {
+        return "ifc";
+    }
+
+    /**
+     * Checks if the specified year is a leap year.
+     * <p>
+     * A leap-year is a year of a longer length than normal.
+     * The exact meaning is determined by the chronology according to the following constraints.
+     * <ul>
+     * <li>a leap-year must imply a year-length longer than a non leap-year.
+     * <li>a chronology that does not support the concept of a year must return false.
+     * </ul>
+     *
+     * @param year the proleptic-year to check, not validated for range
+     * @return true if the year is a leap year
+     */
+    @Override
+    public boolean isLeapYear (final long year) {
+        return ((year & 3) == 0) && ((year % 100) != 0 || (year % 400) == 0);
+    }
+
+    /**
+     * Calculates the proleptic-year given the era and year-of-era.
+     * <p>
+     * The International Fixed calendar only knows common era years, thus negative years are invalid.
+     * <p>
+     * If the chronology makes active use of eras, such as {@code JapaneseChronology}
+     * then the year-of-era will be validated against the era.
+     * For other chronologies, validation is optional.
+     *
+     * @param era       the era of the correct type for the chronology, not null
+     * @param yearOfEra the chronology year-of-era
+     * @return the proleptic-year
+     * @throws DateTimeException  if unable to convert to a proleptic-year,
+     *                            such as if the year is invalid for the era
+     * @throws ClassCastException if the {@code era} is not of the correct type for the chronology
+     */
+    @Override
+    public int prolepticYear (final Era era, final int yearOfEra) {
+        if (!(era instanceof InternationalFixedEra)) {
+            throw new ClassCastException ("Era must be InternationalFixedEra");
+        }
+
+        if (1 > yearOfEra) {
```

Comment:
I get where you're going here, but it's still backwards.  Further, the exception message doesn't really match the condition you check (0 is positive for integers).  I also recommend not putting any special emphasis on `"must"`.